### PR TITLE
test: add a test for fetching long assets with chunkwise certification

### PR DESCRIPTION
**Pull request by przydatek**
Add an integration test for boundary nodes, testing the feature of fetching long assets with chunk-wise certification.

As a side-effect, refactor `asset_canister`-test-helper, to enable deployment with custom WASMs.

### CODE REVIEW

File: rs/tests/boundary_nodes/bn_integration_on_playnet_test.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ fn main() -> Result<()> {
                 .add_test(systest!(api_sync_call_test))
                 .add_test(systest!(api_canister_read_state_test))
                 .add_test(systest!(api_subnet_read_state_test))
-                .add_test(systest!(asset_canister_test))
+                .add_test(systest!(legacy_asset_canister_test))
```
Comment (r-birkner) [inline rs/tests/boundary_nodes/bn_integration_on_playnet_test.rs:37]: Here, we should also add the `long_asset_canister_test`. It is basically the same test as `bn_integration_test`, but it uses real certificates.

**Review by r-birkner — APPROVED** [review]

File: rs/tests/boundary_nodes/bn_integration_on_playnet_test.rs (hunk 1 above)
Comment (przydatek) [reply rs/tests/boundary_nodes/bn_integration_on_playnet_test.rs:37]: Done.

**Review by przydatek — COMMENTED** [review]
Thanks!

**Review by marko-k0 — APPROVED** [review]
